homeassistant/components/sensor/filter.py

```diff
@@ -207,11 +262,20 @@ class Filter(object):
 
     def __init__(self, name, window_size=1, precision=None, entity=None):
         """Initialize common attributes."""
-        self.states = deque(maxlen=window_size)
+        if isinstance(window_size, int):
+            self.states = deque(maxlen=window_size)
+        else:
+            self.states = deque(maxlen=0)
         self.precision = precision
         self._name = name
         self._entity = entity
         self._skip_processing = False
+        self._window_size = window_size
+
+    @property
+    def window_size(self):
```

Comment:
I think that we should have 2 vars. window size and time window ?

Reply:
That would be a very bad decision, there is only 1 dimension (window size) that can be described using different units (time vs #events). You don't keep 2 temperatures F and C...

The issue at hand is that the conversion between these units is non linear and requires a query (converting the #events into time) which is quite expensive.

Reply:
We actually have another variable to describe the unit of the window size. So if you want to make the comparison with temperature, let's add a variable that describes the type.

Reply:
don't really see the need for it, since we can already check that (is actually used throughout the code):

isinstance(filt.window_size, int) -> #events
isinstance(filt.window_size, timedelta) -> time

but I'm fine adding such method:

Would this work for you?
```python
def window_size_unit(self):
  if isinstance(self.window_size, int):
    return WINDOW_SIZE_UNIT_TIME
  else:
    return WINDOW_SIZE_UNIT_NUMBER_EVENTS
```

Reply:
Why not assign it as an instance variable inside the if;else block inside the constructor of Filter.